### PR TITLE
Prims algorithm

### PR DESCRIPTION
**Pull request by Nananaw**
implemented prims algorithm for AdjacencyMatrixGraph and AdjacencyListGraph

### CODE REVIEW

File: Algorithm Codes/Graph/AdjacencyListGraph.py [hunk 1]
```diff
@@ -150,6 +150,34 @@ def floyd_warshall(self):
...
+    def prims(self):
+        nodes_left = self.n
+        visited = [False for i in range(self.n)]
+        visited[0] = True
+        edges = []
+
+        heap = []
+
+        curr = self.adj[0].head
+        while(curr is not None):
```
Comment (seungholee-dev) [inline Algorithm Codes/Graph/AdjacencyListGraph.py:163]:
```suggestion
        while(curr):
```

A minor change according to PEP8 :)

File: Algorithm Codes/Graph/AdjacencyListGraph.py [hunk 2]
```diff
@@ -150,6 +150,34 @@ def floyd_warshall(self):
...
+            heapq.heappush(heap, (curr.weight, (0, curr.node)))
+            curr = curr.next
+        
+        while(nodes_left > 0):
+            (_, (u, v)) = heapq.heappop(heap)
```
Comment (seungholee-dev) [inline Algorithm Codes/Graph/AdjacencyListGraph.py:168]:
```suggestion
            _, (u, v) = heapq.heappop(heap)
```

A minor change according to PEP8 :)

File: Algorithm Codes/Graph/AdjacencyListGraph.py [hunk 3]
```diff
@@ -150,6 +150,34 @@ def floyd_warshall(self):
...
+            edges.append((u, v))
+            visited[v] = True
+            nodes_left = nodes_left - 1
+
+            curr = self.adj[v].head
+            while(curr is not None):
```
Comment (seungholee-dev) [inline Algorithm Codes/Graph/AdjacencyListGraph.py:174]:
```suggestion
            while(curr):

```


A minor change according to PEP8 :)

File: Algorithm Codes/Graph/AdjacencyListGraph.py [hunk 4]
```diff
@@ -150,6 +150,34 @@ def floyd_warshall(self):
...
+        curr = self.adj[0].head
+        while(curr is not None):
+            heapq.heappush(heap, (curr.weight, (0, curr.node)))
+            curr = curr.next
+        
+        while(nodes_left > 0):
+            (_, (u, v)) = heapq.heappop(heap)
+            edges.append((u, v))
+            visited[v] = True
+            nodes_left = nodes_left - 1
```
Comment (seungholee-dev) [inline Algorithm Codes/Graph/AdjacencyListGraph.py:171]:
```suggestion
            nodes_left -= 1

```

This is more of a suggestion for making it looking even more concise. What do you think?! :D

**Review by seungholee-dev — CHANGES_REQUESTED** [review]
Great code! 
Thanks for going through Prim's :) 

Really loved that you used min_heap version which is optimzed version haha
Just left some minor improvements we can strive for! 
Do let me know what do you think!

**Comment by seungholee-dev** [comment]
I have left some comments under your review :)  Just a minor improvements. Feel free to share your thoughts! Thank you so much!

File: Algorithm Codes/Graph/AdjacencyListGraph.py (hunk 1 above)
Comment (Nananaw) [reply Algorithm Codes/Graph/AdjacencyListGraph.py:163]: https://peps.python.org/pep-0008/#programming-recommendations
says to use is not none

File: Algorithm Codes/Graph/AdjacencyListGraph.py (hunk 2 above)
Comment (Nananaw) [reply Algorithm Codes/Graph/AdjacencyListGraph.py:168]: looks good

File: Algorithm Codes/Graph/AdjacencyListGraph.py (hunk 3 above)
Comment (Nananaw) [reply Algorithm Codes/Graph/AdjacencyListGraph.py:174]: commented above about this

File: Algorithm Codes/Graph/AdjacencyListGraph.py (hunk 4 above)
Comment (Nananaw) [reply Algorithm Codes/Graph/AdjacencyListGraph.py:171]: sounds good

File: Algorithm Codes/Graph/AdjacencyListGraph.py (hunk 1 above)
Comment (seungholee-dev) [reply Algorithm Codes/Graph/AdjacencyListGraph.py:163]: Oh you are right! 
I will drop my suggestion :) 

Thanks!

Comment (seungholee-dev) [reply Algorithm Codes/Graph/AdjacencyListGraph.py:163]: I noticed your way should be suitable for this case.
I did a quick research and this is also filtering `[ ]` and `0` and others. So `is not None` should be right :) 

I will just remove the parenthesis

**Comment by seungholee-dev** [comment]
Merging this!  :) 

Thank you, the code looks amazing!

**Review by seungholee-dev — APPROVED** [review]

**Pull request merged**